lib/susedistribution.pm

```diff
@@ -203,10 +203,8 @@ sub init_consoles {
             $s390_params =~ s,\@S390_HOST\@,$s390_host,g;
             set_var("S390_NETWORK_PARAMS", $s390_params);
 
-            my @s390_params = split / /, $s390_params;
-            my %s390_params = map { split "=" } @s390_params;
-
-            $hostname = $s390_params{Hostname};
+            $hostname = $s390_params;
+            $hostname =~ s/.*Hostname=(\S*)\s.*/$1/;
```

Comment:
I had the same thoughts.
But instead of all this copy and substitute I would just use:
```perl
($hostname) = $s390_params =~ /Hostname=(\S+)/;
```
`\S+` instead of `\S*` to be sure and avoid greedy `.*` at the beginning and at the end. This way `$s390_params` is even preserved if needed.